FullTextSearch/Services.cs

```diff
@@ -0,0 +1,41 @@
+using FullTextSearch.Application.DocumentsReader;
+using FullTextSearch.Application.InvertedIndex;
+using FullTextSearch.Application.Searchers;
+using FullTextSearch.Application.WordsProcessors;
+using Microsoft.Extensions.DependencyInjection;
+using Porter2Stemmer;
+
+namespace FullTextSearch;
+
+internal static class Services
+{
+    public static ServiceProvider ConfigureServices()
+    {
+        var serviceCollector = new ServiceCollection();
+
+        serviceCollector.AddSingleton<IPorter2Stemmer, EnglishPorter2Stemmer>();
```

Comment:
What are the advantages of singleton class?
read about Addtransient and Addscope.
When is it dangerous to use singleton classes?